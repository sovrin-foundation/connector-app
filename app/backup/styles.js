// @flow
import { StyleSheet, Dimensions, Platform } from 'react-native'
import {
  PADDING_HORIZONTAL,
  PADDING_VERTICAL,
  isBiggerThanShortDevice,
  color,
  grey,
} from '../common/styles/constant'

const { width, height } = Dimensions.get('window')
const SPACE_FILLER = 50
const amountSpacing = isBiggerThanShortDevice ? PADDING_VERTICAL * 2 : 0
const submitButtonHeight = isBiggerThanShortDevice ? 63 : 43
const showRecoveryButtonHeight = isBiggerThanShortDevice ? 150 : 100
const inputBoxHeight =
  isBiggerThanShortDevice || Platform.OS === 'ios' ? 137 : 40
export const chatBubbleTextOffset = 40
const styles = StyleSheet.create({
  tabContainer: {
    maxHeight: 60,
  },
  backgroundImage: {
    width: '100%',
    flex: 1,
    position: 'absolute',
  },
  headerCloseIcon: {
    marginRight: PADDING_HORIZONTAL,
    marginBottom: PADDING_VERTICAL,
    alignSelf: 'flex-end',
  },
  headerIcon: {
    marginRight: isBiggerThanShortDevice ? 5 : 0,
    marginTop: isBiggerThanShortDevice ? 15 : 0,
    alignSelf: 'flex-end',
  },
  headerBackIcon: {
    marginLeft: isBiggerThanShortDevice ? 5 : 0,
    marginTop: isBiggerThanShortDevice ? 15 : 5,
    alignSelf: 'flex-start',
  },
  headerSpacer: {
    height: '100%',
    width: isBiggerThanShortDevice ? SPACE_FILLER : SPACE_FILLER * 3 / 4,
  },
  showRecoveryPassphrase: {
    width: '86%',
    height: showRecoveryButtonHeight,
    borderRadius: 5,
    alignSelf: 'center',
    backgroundColor: color.bg.tertiary.color,
    elevation: 7,
    marginBottom: 60,
  },
  genRecovery: {
    backgroundColor: color.bg.eleventh.color,
  },
  genRecoveryHeader: {
    width: '100%',
  },
  genRecoveryPhraseContainer: {
    position: 'absolute',
    top: 20,
    left: chatBubbleTextOffset / 2,
    paddingHorizontal: '7%',
  },
  genRecoveryPhraseLoadingContainer: {
    padding: 40,
    position: 'absolute',
    top: 0,
  },
  genRecoveryPhrase: {
    textAlign: 'center',
    color: color.bg.eleventh.color,
    fontSize: isBiggerThanShortDevice ? 20 : 18,
    lineHeight: isBiggerThanShortDevice ? 30 : 28,
    fontWeight: '900',
  },
  genRecoveryText: {
    marginBottom: isBiggerThanShortDevice ? 10 : '2%',
    alignSelf: 'center',
    fontSize: isBiggerThanShortDevice ? 22 : 20,
    fontWeight: '600',
  },
  genRecoveryMessage: {
    paddingHorizontal: 20,
    alignSelf: 'center',
    fontSize: isBiggerThanShortDevice ? 18 : 16,
    lineHeight: isBiggerThanShortDevice ? 22 : 20,
    fontWeight: '500',
    marginBottom: isBiggerThanShortDevice ? 20 : 5,
  },
  genRecoverySecondMessage: {
    paddingHorizontal: 20,
    alignSelf: 'center',
    fontSize: isBiggerThanShortDevice ? 18 : 16,
    lineHeight: isBiggerThanShortDevice ? 22 : 20,
    fontWeight: 'bold',
  },
  genRecoverySmallMessage: {
    fontSize: 14,
    lineHeight: 17,
    fontWeight: 'bold',
    marginBottom: isBiggerThanShortDevice ? 15 : '3%',
  },
  verifyMainContainer: {
    flex: 1,
    backgroundColor: color.bg.twelfth.color,
  },
  verifyMainText: {
    paddingHorizontal: 20,
    fontSize: 18,
    lineHeight: 22,
    fontWeight: '500',
    marginTop: isBiggerThanShortDevice ? 40 : 20,
    marginBottom: isBiggerThanShortDevice ? 40 : 20,
  },
  exportBackupMainText: {
    paddingHorizontal: 10,
    fontSize: isBiggerThanShortDevice ? 18 : 16,
    lineHeight: isBiggerThanShortDevice ? 22 : 20,
    fontWeight: '500',
  },
  inputBox: {
    marginBottom: 24,
    marginRight: 20,
    marginLeft: 20,
    height: inputBoxHeight,
    backgroundColor: 'rgba(0,0,0,0.33)',
    color: 'white',
    padding: 10,
    textAlignVertical: 'top',
  },
  submitButton: {
    marginLeft: '2%',
    marginRight: '2%',
    marginBottom: 15,
    borderRadius: 5,
    shadowColor: grey,
    shadowOffset: {
      width: 1,
      height: 1,
    },
    shadowRadius: 5,
    shadowOpacity: 0.3,
    elevation: 7,
    height: isBiggerThanShortDevice ? 53 : 43,
  },
  backgroundImageVerify: {
    flex: 1,
    position: 'absolute',
  },
  exportBackup: {
    backgroundColor: color.bg.thirteenth.color,
  },
  wrapper: {
    paddingHorizontal: PADDING_HORIZONTAL,
  },
  title: {
    fontWeight: '600',
    lineHeight: isBiggerThanShortDevice ? 27 : 22,
    fontSize: isBiggerThanShortDevice ? 22 : 20,
    marginBottom: isBiggerThanShortDevice ? 20 : 10,
    width: '100%',
  },
  exportBackupTitle: {
    fontWeight: '600',
    lineHeight: isBiggerThanShortDevice ? 27 : 22,
    fontSize: isBiggerThanShortDevice ? 22 : 20,
    marginBottom: isBiggerThanShortDevice ? 30 : 10,
    marginTop: isBiggerThanShortDevice ? 10 : 0,
    maxWidth: 290,
  },
  exportBackupText: {
    marginBottom: 20,
    backgroundColor: color.bg.thirteenth.color,
    alignSelf: 'center',
    fontSize: 22,
  },
  exportBackupMessage: {
    backgroundColor: color.bg.thirteenth.color,
    alignSelf: 'center',
    fontSize: 18,
  },
  exportBackupFile: {
    fontSize: 15,
    lineHeight: 18,
    fontWeight: '900',
  },
  exportBackupSmallMessage: {
    fontSize: 14,
    maxWidth: 300,
    lineHeight: 17,
    marginBottom: isBiggerThanShortDevice ? 30 : 10,
  },
  backupComplete: {
    backgroundColor: color.bg.fourteenth.color,
  },
  backupCompleteHeader: {
    width: '100%',
  },
  backupCompleteText: {
    fontSize: 18,
    fontWeight: '500',
    lineHeight: 22,
  },
  alignItemsCenter: {
    marginBottom: 6,
    marginLeft: '5%',
    marginRight: '5%',
  },
  verticalSpacing: {
    paddingTop: amountSpacing,
  },
  customButtonColor: {
    fontWeight: '600',
    fontSize: 18,
  },
  lockIconImage: {
    paddingTop: isBiggerThanShortDevice ? 40 : '2%',
  },
  imageIcon: {
    resizeMode: 'stretch',
    width: '100%',
    marginBottom: isBiggerThanShortDevice ? 20 : 0,
  },
  imageIconEncryptFile: {
    resizeMode: 'contain',
    width: isBiggerThanShortDevice ? '100%' : '80%',
    marginBottom: isBiggerThanShortDevice ? 20 : 0,
  },
})

export default styles
