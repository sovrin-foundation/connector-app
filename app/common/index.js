// @flow
export * from './button-constants'
export * from './message-constants'
export * from './route-constants'
export * from './secure-storage-constants'
export function noop() {}
