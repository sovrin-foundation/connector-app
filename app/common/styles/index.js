// @flow
export * from './constant'
export * from './tab-constants'
