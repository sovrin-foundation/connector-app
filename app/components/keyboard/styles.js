// @flow

import { StyleSheet, Dimensions, Platform } from 'react-native'
import {
  isBiggerThanShortDevice,
  deviceHeight,
  MEDIUM_DEVICE,
} from '../../common/styles/constant'
import { PUSH_NOTIFICATION_PERMISSION } from '../../push-notification/type-push-notification'

const ROW_MARGIN = isBiggerThanShortDevice ? 36 : 15
const FIRST_ROW_MARGIN = isBiggerThanShortDevice ? 24 : 8

const ROW_HEIGHT =
  Platform.OS === 'ios' ? 36 : deviceHeight < MEDIUM_DEVICE ? 27 : 36

const FONT_SIZE = isBiggerThanShortDevice ? 28 : 26
export const INPUT_RANGE = [0, 0.5, 1]
export const OUTPUT_RANGE =
  Platform.OS === 'ios'
    ? isBiggerThanShortDevice ? [28, 29, 30] : [26, 30, 34]
    : deviceHeight < MEDIUM_DEVICE ? [28, 30, 32] : [26, 30, 34]

export const styles = StyleSheet.create({
  container: {
    marginHorizontal: 32, // TODO: update style guide to reflect new constant
    marginBottom: 5,
  },
  firstRow: {
    height: ROW_HEIGHT,
    marginTop: FIRST_ROW_MARGIN,
  },
  row: {
    height: ROW_HEIGHT,
    marginTop: ROW_MARGIN,
  },
  number: {
    fontSize: FONT_SIZE,
    fontWeight: '500', // TODO: update style guide to reflect new constant
  },
})
