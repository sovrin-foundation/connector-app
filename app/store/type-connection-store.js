// @flow
import type { InvitationPayload } from '../invitation/type-invitation'
import type { CustomError } from '../common/type-common'

export type MyPairwiseInfo = {
  myPairwiseDid: string,
  myPairwiseVerKey: string,
  myPairwiseAgentDid: string,
  myPairwiseAgentVerKey: string,
  myPairwisePeerVerKey: string,
  senderDID: string,
}

export type Connection = {
  identifier: string,
  logoUrl: string,
  senderEndpoint: string,
  size?: number,
  senderName?: string,
  vcxSerializedConnection: string,
} & MyPairwiseInfo

export const DELETE_CONNECTION = 'DELETE_CONNECTION'

export type DeleteConnectionEventAction = {
  type: typeof DELETE_CONNECTION,
  senderDID: string,
}

export type Connections = { [senderDID: string]: Connection }

export type ConnectionStore = {
  // TODO:PS Add specific keys in connection store
  [string]: any,
  data: ?Connections,
}

export const DELETE_CONNECTION_SUCCESS = 'DELETE_CONNECTION_SUCCESS'

export const DELETE_CONNECTION_FAILURE = 'DELETE_CONNECTION_FAILURE'

export type DeleteConnectionSuccessEventAction = {
  type: typeof DELETE_CONNECTION_SUCCESS,
  filteredConnections: Connections,
}

export type DeleteConnectionFailureEventAction = {
  type: typeof DELETE_CONNECTION_FAILURE,
  connection: Connection,
  error: CustomError,
}

export const NEW_CONNECTION = 'NEW_CONNECTION'

export type NewConnectionAction = {
  type: typeof NEW_CONNECTION,
  connection: {
    identifier: string,
    logoUrl?: ?string,
  } & InvitationPayload,
}

export const STORAGE_KEY_THEMES = 'STORAGE_KEY_THEMES'

export type ConnectionThemes = {
  [string]: {
    primary: string,
    secondary: string,
  },
}

export const HYDRATE_CONNECTION_THEMES = 'HYDRATE_CONNECTION_THEMES'
