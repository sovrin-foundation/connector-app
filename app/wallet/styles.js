// @flow
import { StyleSheet, Dimensions } from 'react-native'
import {
  PADDING_HORIZONTAL,
  PADDING_VERTICAL,
  isBiggerThanShortDevice,
  color,
} from '../common/styles/constant'

const { width } = Dimensions.get('window')
const SPACE_FILLER = 50
const amountSpacing = isBiggerThanShortDevice ? PADDING_VERTICAL * 2 : 0
const ctaButtonHeight = isBiggerThanShortDevice ? 63 : 43

const styles = StyleSheet.create({
  headerCloseIcon: {
    marginTop: PADDING_VERTICAL,
  },
  ctaButton: {
    flex: -1,
    height: ctaButtonHeight,
    borderRadius: 5,
    backgroundColor: color.bg.eighth.color,
  },
  alignItemsCenter: {
    marginBottom: 6,
    marginHorizontal: '5%',
  },
  verticalSpacing: {
    flexDirection: 'column',
    alignContent: 'center',
    justifyContent: 'space-around',
  },
})

export default styles
