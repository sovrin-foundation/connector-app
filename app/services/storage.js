// @flow
import { AsyncStorage } from 'react-native'
import { call, select } from 'redux-saga/effects'
import RNSensitiveInfo from 'react-native-sensitive-info'
import {
  setWalletItem,
  getWalletItem,
  deleteWalletItem,
  updateWalletItem,
  vcxShutdown,
} from '../bridge/react-native-cxs/RNCxs'
import { captureError } from './error/error-handler'
import { IN_RECOVERY } from '../lock/type-lock'
import { noop } from '../common'

const storageName = {
  sharedPreferencesName: 'ConnectMeSharedPref',
  keychainService: 'ConnectMeKeyChain',
}

// SECURE STORAGE
export const secureSet = (key: string, data: string) =>
  RNSensitiveInfo.setItem(key, data, storageName)

export const secureGet = async (key: string) => {
  const data = await RNSensitiveInfo.getItem(key, storageName)

  return data
}

export const secureGetAll = () => RNSensitiveInfo.getAllItems(storageName)

export const secureDelete = (key: string) =>
  RNSensitiveInfo.deleteItem(key, storageName)

export const getStackTrace = () => {
  var obj = {}
  Error.captureStackTrace(obj, getStackTrace)
  return obj.stack
}

// WALLET STORAGE
export const walletSet = async (key: string, data: string) => {
  try {
    await setWalletItem(key, data)
  } catch (err) {
    captureError(err)
    try {
      // if add item fails, try update item
      await walletUpdate(key, data)
    } catch (e) {
      captureError(e)
      // need to think about what happens if storage fails
      console.log(`Storage fails: key: ${key}, Error: ${e}`)
    }
  }
}

export const walletGet = async (key: string) => {
  try {
    const data: string = await getWalletItem(key)
    return data
  } catch (e) {
    captureError(e)
    console.log(`walletGet: key: ${key}, Error: ${e}`)
    return null
  }
}

export const walletDelete = async (key: string) => {
  try {
    const del = await deleteWalletItem(key)
    return del
  } catch (e) {
    captureError(e)
  }
}

export async function walletUpdate(key: string, data: string) {
  try {
    await updateWalletItem(key, data)
  } catch (err) {
    captureError(err)
    console.log(
      'walletUpdate error :: key: ' + key + ' :: data: ' + data + ' :: err: ',
      err
    )
  }
}

// NON-SECURE STORAGE
export const safeSet = (key: string, data: string) =>
  AsyncStorage.setItem(key, data)

export const safeGet = (key: string) => AsyncStorage.getItem(key)

export const safeDelete = (key: string) => AsyncStorage.removeItem(key)

export const safeMultiRemove = (keys: string[]) =>
  AsyncStorage.multiRemove(keys)

// GET ITEM USED FOR HYDRATION METHODS
export const getHydrationItem = async (key: string) => {
  let inRecovery: string = await safeGet(IN_RECOVERY)

  if (inRecovery === 'true') {
    const walletItem: string | null = await walletGet(key)

    if (walletItem) {
      // put items inside secure storage in background
      // not using await syntax, because we want this in background
      secureSet(key, walletItem)
        .then(noop)
        .catch(noop)
    }

    return walletItem
  } else {
    const secureItem = secureGet(key)

    return secureItem
  }
}
