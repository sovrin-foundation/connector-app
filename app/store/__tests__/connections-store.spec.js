// @flow
import renderer from 'react-test-renderer'
import * as matchers from 'redux-saga-test-plan/matchers'
import { expectSaga } from 'redux-saga-test-plan'
import { throwError } from 'redux-saga-test-plan/providers'

import connectionReducer, {
  saveNewConnection,
  saveNewConnectionSuccess,
  saveNewConnectionFailed,
  connectionMapper,
  deleteConnectionSuccess,
  deleteConnectionOccurredSaga,
  deleteConnectionAction,
  persistThemes,
  hydrateThemes,
  removePersistedThemes,
  hydrateConnectionThemes,
} from '../connections-store'
import { bubbleSize } from '../../common/styles'
import {
  successConnectionData,
  connections,
  userOneTimeInfo,
  configStoreNotHydratedInstalledVcxInit,
  getStore,
  connectionThemes,
} from '../../../__mocks__/static-data'
import { safeGet, safeSet, secureSet, secureGet } from '../../services/storage'
import { CONNECTIONS } from '../../common'
import { deleteConnection } from '../../bridge/react-native-cxs/RNCxs'
import { STORAGE_KEY_THEMES } from '../type-connection-store'

describe('Mapper', () => {
  it('connectionMapper should return proper object', () => {
    const connection = {
      identifier: '3nj819kkjywdppuje79',
      logoUrl: 'https://test-agengy.com/logo',
      size: bubbleSize.XL,
      name: 'test',
      senderDID: '123819kkjywdppuj987',
      senderEndpoint: 'https://test-endpoint.com',
      remoteConnectionId: '5iZiu2aLYrQXSdon123456',
    }
    const tree = connectionMapper(connection)
    expect(tree).toMatchSnapshot()
  })
})

describe('connections should update correctly', () => {
  let initialState = {
    data: null,
  }
  const newConnection = {
    identifier: '6789012345678906789012',
    name: 'test',
  }

  beforeAll(() => {
    // get initial state without any action
    initialState = connectionReducer(undefined, { type: 'NO_ACTION' })
  })

  it('should receive new connection request', () => {
    const expectedState = {
      ...initialState,
      isFetching: true,
      isPristine: false,
    }
    const actualState = connectionReducer(
      initialState,
      saveNewConnection(newConnection)
    )
    expect(actualState).toMatchObject(expectedState)
  })

  it('match delete connection success', () => {
    expect(
      connectionReducer(initialState, deleteConnectionSuccess(connections))
    ).toMatchSnapshot()
  })

  it('saga:deleteConnectionOccurredSaga, success', () => {
    const connection = successConnectionData.newConnection
    const stateWithConnection = {
      connections: {
        data: {
          [connection.identifier]: connection,
        },
      },
      user: {
        userOneTimeInfo,
      },
      config: configStoreNotHydratedInstalledVcxInit,
    }

    return expectSaga(
      deleteConnectionOccurredSaga,
      deleteConnectionAction(connection.senderDID)
    )
      .withState(stateWithConnection)
      .provide([
        [matchers.call.like({ fn: deleteConnection }), true],
        [matchers.call.fn(secureSet, CONNECTIONS, '{}'), true],
      ])
      .call.like({ fn: deleteConnection })
      .call(secureSet, CONNECTIONS, '{}')
      .put(deleteConnectionSuccess({}))
      .run()
  })

  it('should update connections and store new connection properly', () => {
    const expectedState = {
      ...initialState,
      isFetching: false,
      data: {
        ...initialState.data,
        [newConnection.identifier]: newConnection,
      },
    }
    const actualState = connectionReducer(
      initialState,
      saveNewConnectionSuccess(newConnection)
    )
    expect(actualState).toMatchObject(expectedState)
  })

  it('should fail if new connection is not stored', () => {
    const error = { code: '1234', message: 'new connection failed' },
      expectedState = {
        ...initialState,
        isFetching: false,
        error,
      }
    const actualState = connectionReducer(
      initialState,
      saveNewConnectionFailed(error)
    )
    expect(actualState).toMatchObject(expectedState)
  })

  it('should reset connection store, if RESET action is raised', () => {
    const afterNewConnection = connectionReducer(
      initialState,
      saveNewConnection(newConnection)
    )
    expect(
      connectionReducer(afterNewConnection, { type: 'RESET' })
    ).toMatchSnapshot()
  })

  it('saga:persistThemes, success', async () => {
    const stateWithThemes = getStore().getState()
    const result = await expectSaga(persistThemes)
      .withState(stateWithThemes)
      .provide([[matchers.call.like({ fn: secureSet }), true]])
      .run()
    expect(result).toMatchSnapshot()
  })

  it('saga:hydrateThemes, success', async () => {
    const result = await expectSaga(hydrateThemes)
      .provide([
        [
          matchers.call.fn(secureGet, STORAGE_KEY_THEMES),
          JSON.stringify(connectionThemes),
        ],
      ])
      .run()
    expect(result).toMatchSnapshot()
  })

  it('saga:removePersistedThemes, success', async () => {
    const result = await expectSaga(removePersistedThemes).run()
    expect(result).toMatchSnapshot()
  })

  it('ACTION:HYDRATE_CONNECTION_THEMES', () => {
    const afterThemeHydrationState = connectionReducer(
      initialState,
      hydrateConnectionThemes(connectionThemes)
    )
    expect(afterThemeHydrationState).toMatchSnapshot()
  })
})
