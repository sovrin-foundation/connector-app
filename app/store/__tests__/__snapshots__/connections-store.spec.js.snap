// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mapper connectionMapper should return proper object 1`] = `
Object {
  "identifier": "3nj819kkjywdppuje79",
  "logoUrl": "https://test-agengy.com/logo",
  "name": "test",
  "remoteConnectionId": "5iZiu2aLYrQXSdon123456",
  "senderDID": "123819kkjywdppuj987",
  "senderEndpoint": "https://test-endpoint.com",
  "senderName": "Unknown",
  "size": 120,
}
`;

exports[`connections should update correctly ACTION:HYDRATE_CONNECTION_THEMES 1`] = `
Object {
  "claimMap": null,
  "connectionThemes": Object {
    "default": Object {
      "primary": "rgba(119, 119, 119, 1.0)",
      "secondary": "rgba(119, 119, 119, 0.6)",
    },
    "http://cr0ybot.github.io/ingress-logos/ingress.png": Object {
      "primary": "rgba(100,200,100,1)",
      "secondary": "rgba(100,200,100,0.6)",
    },
  },
  "data": Object {},
  "error": Object {
    "code": "",
    "message": "",
  },
  "hydrated": false,
  "isFetching": false,
  "isPristine": true,
}
`;

exports[`connections should update correctly match delete connection success 1`] = `
Object {
  "claimMap": null,
  "connectionThemes": Object {
    "default": Object {
      "primary": "rgba(119, 119, 119, 1.0)",
      "secondary": "rgba(119, 119, 119, 0.6)",
    },
  },
  "data": Object {
    "3nj819kkjywdppuje86": Object {
      "identifier": "3nj819kkjywdppuje86",
      "logoUrl": "https://logourl.com/logo.png",
      "myPairwiseAgentDid": "myPairwiseAgentDID",
      "myPairwiseAgentVerKey": "myPairwiseAgentVerKey",
      "myPairwiseDid": "pairwiseIdentifier1",
      "myPairwisePeerVerKey": "senderVerificationKey1",
      "myPairwiseVerKey": "pairwiseVerificationKey1",
      "name": "Test Connection2",
      "remoteConnectionId": "70075yyojywdppuje79",
      "requestId": "requestId1",
      "senderAgencyDetail": Object {
        "DID": "senderDID1",
        "endpoint": "endpoint",
        "verKey": "senderVerificationKey1",
      },
      "senderAgentKeyDelegationProof": Object {
        "agentDID": "senderAgentDID1",
        "agentDelegatedKey": "agentDelegatedKey",
        "signature": "signature",
      },
      "senderDID": "senderDID",
      "senderDetail": Object {
        "DID": "senderDID1",
        "agentKeyDlgProof": Object {
          "agentDID": "senderAgentDID1",
          "agentDelegatedKey": "agentDelegatedKey",
          "signature": "signature",
        },
        "logoUrl": "http://testissuer.com/logoUrl.png",
        "name": "sender1",
        "verKey": "senderVerificationKey1",
      },
      "senderEndpoint": "endpoint",
      "senderLogoUrl": "lu",
      "senderName": "senderName",
      "senderVerificationKey": "sVk",
      "size": 100,
      "targetName": "target name",
      "vcxSerializedConnection": "",
    },
    "4ej819kkjywdppuje79": Object {
      "identifier": "4ej819kkjywdppuje79",
      "logoUrl": "https://logourl.com/logo.png",
      "myPairwiseAgentDid": "myPairwiseAgentDID",
      "myPairwiseAgentVerKey": "myPairwiseAgentVerKey",
      "myPairwiseDid": "pairwiseIdentifier1",
      "myPairwisePeerVerKey": "senderVerificationKey1",
      "myPairwiseVerKey": "pairwiseVerificationKey1",
      "name": "Test Connection1",
      "remoteConnectionId": "70075yyojywdppuje79",
      "requestId": "requestId1",
      "senderAgencyDetail": Object {
        "DID": "senderDID1",
        "endpoint": "endpoint",
        "verKey": "senderVerificationKey1",
      },
      "senderAgentKeyDelegationProof": Object {
        "agentDID": "senderAgentDID1",
        "agentDelegatedKey": "agentDelegatedKey",
        "signature": "signature",
      },
      "senderDID": "senderDID",
      "senderDetail": Object {
        "DID": "senderDID1",
        "agentKeyDlgProof": Object {
          "agentDID": "senderAgentDID1",
          "agentDelegatedKey": "agentDelegatedKey",
          "signature": "signature",
        },
        "logoUrl": "http://testissuer.com/logoUrl.png",
        "name": "sender1",
        "verKey": "senderVerificationKey1",
      },
      "senderEndpoint": "endpoint",
      "senderLogoUrl": "lu",
      "senderName": "senderName",
      "senderVerificationKey": "sVk",
      "size": 100,
      "targetName": "target name",
      "vcxSerializedConnection": "",
    },
    "7fj819kkjywdppuje34": Object {
      "identifier": "7fj819kkjywdppuje34",
      "logoUrl": "https://logourl.com/logo.png",
      "myPairwiseAgentDid": "myPairwiseAgentDID",
      "myPairwiseAgentVerKey": "myPairwiseAgentVerKey",
      "myPairwiseDid": "pairwiseIdentifier1",
      "myPairwisePeerVerKey": "senderVerificationKey1",
      "myPairwiseVerKey": "pairwiseVerificationKey1",
      "name": "Test Connection3",
      "remoteConnectionId": "70075yyojywdppuje79",
      "requestId": "requestId1",
      "senderAgencyDetail": Object {
        "DID": "senderDID1",
        "endpoint": "endpoint",
        "verKey": "senderVerificationKey1",
      },
      "senderAgentKeyDelegationProof": Object {
        "agentDID": "senderAgentDID1",
        "agentDelegatedKey": "agentDelegatedKey",
        "signature": "signature",
      },
      "senderDID": "senderDID",
      "senderDetail": Object {
        "DID": "senderDID1",
        "agentKeyDlgProof": Object {
          "agentDID": "senderAgentDID1",
          "agentDelegatedKey": "agentDelegatedKey",
          "signature": "signature",
        },
        "logoUrl": "http://testissuer.com/logoUrl.png",
        "name": "sender1",
        "verKey": "senderVerificationKey1",
      },
      "senderEndpoint": "endpoint",
      "senderLogoUrl": "lu",
      "senderName": "senderName",
      "senderVerificationKey": "sVk",
      "size": 100,
      "targetName": "target name",
      "vcxSerializedConnection": "",
    },
  },
  "error": Object {
    "code": "",
    "message": "",
  },
  "hydrated": false,
  "isFetching": false,
  "isPristine": true,
}
`;

exports[`connections should update correctly saga:hydrateThemes, success 1`] = `
Object {
  "call": Array [
    Object {
      "@@redux-saga/IO": true,
      "CALL": Object {
        "args": Array [
          "STORAGE_KEY_THEMES",
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/getHydrationItem",
      },
    },
  ],
}
`;

exports[`connections should update correctly saga:persistThemes, success 1`] = `
Object {
  "call": Array [
    Object {
      "@@redux-saga/IO": true,
      "CALL": Object {
        "args": Array [
          "STORAGE_KEY_THEMES",
          "{\\"default\\":{\\"primary\\":\\"rgba(119, 119, 119, 1.0)\\",\\"secondary\\":\\"rgba(119, 119, 119, 0.6)\\"}}",
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/secureSet",
      },
    },
  ],
  "select": Array [
    Object {
      "@@redux-saga/IO": true,
      "SELECT": Object {
        "args": Array [],
        "selector": "@@redux-saga-test-plan/json/function/getThemes",
      },
    },
  ],
}
`;

exports[`connections should update correctly saga:removePersistedThemes, success 1`] = `
Object {
  "call": Array [
    Object {
      "@@redux-saga/IO": true,
      "CALL": Object {
        "args": Array [
          "STORAGE_KEY_THEMES",
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/secureDelete",
      },
    },
  ],
}
`;

exports[`connections should update correctly should reset connection store, if RESET action is raised 1`] = `
Object {
  "claimMap": null,
  "connectionThemes": Object {
    "default": Object {
      "primary": "rgba(119, 119, 119, 1.0)",
      "secondary": "rgba(119, 119, 119, 0.6)",
    },
  },
  "data": Object {},
  "error": Object {
    "code": "",
    "message": "",
  },
  "hydrated": false,
  "isFetching": false,
  "isPristine": true,
}
`;
